libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Network/Network/OAuth/CSFSalesforceOAuthRefresh.m

```diff
@@ -97,6 +97,12 @@ - (void)oauthCoordinatorDidAuthenticate:(SFOAuthCoordinator *)coordinator authIn
     self.network.account.credentials = coordinator.credentials;
     CSFOAuthTokenRefreshOutput *output = [[CSFOAuthTokenRefreshOutput alloc] initWithCoordinator:coordinator];
     [self finishWithOutput:output error:nil];
+
+    dispatch_async(dispatch_get_main_queue(), ^{
+        [[NSNotificationCenter defaultCenter] postNotificationName:kSFAuthenticationManagerFinishedNotification
```

Comment:
There is a supposed "best practice" going around the interwebs that it is best to always post notifications on the main thread. Although I do not subscribe to this school of thought, I have a good idea of why it has become popular . Another impact of the above change is PostNotification which blocks until all observers have returned would now be async to the OAuthRefresh class.